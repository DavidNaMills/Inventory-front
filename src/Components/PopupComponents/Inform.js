import React from 'react';

const style = {
    height: '100%',
    backgroundColor: 'rgb(237, 237, 237)',
    width: '300px',
    padding: '50px'
}

const Inform = (props) =>{
    const {message, onClose} = props;

    return (
        <div style={style}>
            <p>{message}</p>
            <button onClick={onClose}>Close</button>
        </div>
    )
}

export default Inform;