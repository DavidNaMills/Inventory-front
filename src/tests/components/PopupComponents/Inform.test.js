import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Inform from '../../../Components/PopupComponents/Inform';

const mockStyle = {
    height: '100%',
    backgroundColor: 'rgb(237, 237, 237)',
    width: '300px',
    padding: '50px'
}

const mockProps = {
    message: 'hello world',
    onClose: jest.fn()
};

afterEach(()=>{
    jest.clearAllMocks();
})

describe('<Inform/> popup component test suite', ()=>{
    const wrapper = shallow(<Inform {...mockProps}/>);

    it('matches snapshot', ()=>{
        const comp = renderer.create(<Inform {...mockProps}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <div> with inline styles === mockStyle', ()=>{
        const div = wrapper.find('div');
        expect(div.length).toBe(1);
        expect(div.props().style).toEqual(mockStyle);
    });
    
    it('renders a <p> element containing the message', ()=>{
        const p = wrapper.find('p');
        expect(p.length).toBe(1);
        expect(p.text()).toEqual(mockProps.message);
    });
    
    it('renders a <button/> element and reacts to it being clicked', ()=>{
        const btn = wrapper.find('button');
        expect(btn.length).toBe(1);
        expect(btn.text()).toEqual('Close');
        btn.simulate('click');
        expect(mockProps.onClose).toHaveBeenCalledTimes(1);
    });
});