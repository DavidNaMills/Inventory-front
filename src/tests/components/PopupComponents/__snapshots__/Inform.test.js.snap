// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inform/> popup component test suite matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgb(237, 237, 237)",
      "height": "100%",
      "padding": "50px",
      "width": "300px",
    }
  }
>
  <p>
    hello world
  </p>
  <button
    onClick={[MockFunction]}
  >
    Close
  </button>
</div>
`;
